codegen/cmd/injection-gen/generators/reconciler_controller.go

```diff
@@ -251,7 +261,12 @@ func NewImpl(ctx {{.contextContext|raw}}, r Interface{{if .hasClass}}, classValu
 	)
 
 
-	impl := {{.controllerNewImpl|raw}}(rec, logger, ctrTypeName)
+	options := {{.controllerControllerOptions|raw}}{
+		Logger: logger,
+		WorkQueueName: ctrTypeName,
+		GlobalResyncFilterFunc: func(o interface{}) bool { filterFunc(o) },
```

Comment:
a hacky trick here, wrapping the `filterFunc` in another function closes over the value so we can mutate it down below, passing `filterFunc` directly would pass by value